src/watcher.ts

```diff
@@ -54,6 +64,18 @@ interface FileWatcher {
   watcher: chokidar.FSWatcher;
 }
 
+/**
+ * The minimum time that must elapse after the last file change was detected
+ * before we begin a new run. Also the minimum time between successive runs.
+ *
+ * Note even 0 is a useful value here, because that defers new runs to the next
+ * JS task. This is important because if multiple scripts are watching the same
+ * file that changed, we get a file watcher event for each of them. Without
```

Comment:
Is there a way to see the queue of FS events, consume all pending events synchronously similar to takeRecords in MutationObservers, or see if there are any pending events at all similar to isInputPending()? I'm wondering if in cases of multiple file changes you still get event 1, event 2, run (consume event 1), run (consume event 2).

Reply:
I was wondering how enqueued events were handled, and the queues are run through per event loop turn: https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#phases-overview

Reply:
Cool. Though just for clarity, in the case of Wireit, we have a separate chokidar watcher for each script. So even if a chokidar watcher is batching all changes per JS task, we will still receive a separate event for each script. Hence it's still helpful for us to have our own debounce.